pulsar/internal/connection_pool.go

```diff
@@ -75,13 +75,15 @@ func (p *connectionPool) GetConnection(logicalAddr *url.URL, physicalAddr *url.U
 	p.Lock()
 	conn, ok := p.connections[key]
 	if ok {
-		p.log.Debugf("Found connection in pool key=%s logical_addr=%+v physical_addr=%+v",
+		p.log.Infof("Found connection in pool key=%s logical_addr=%+v physical_addr=%+v",
 			key, conn.logicalAddr, conn.physicalAddr)
 
-		// remove stale/failed connection
-		if conn.closed() {
+		// When the current connection is in a closed state or the broker actively notifies that the
```

Comment:
How does the broker notify that the connection is closed?

Reply:
The action of the broker's notification to close the connection will be processed by the two commands CommandCloseProducer and CommandCloseConsumer. Generally, it may trigger the automatic load balancing of the topic or the unload operation will trigger the active closing of the broker.